### DISCUSSION TITLE
Download a file dynamically created on the net

### DISCUSSION THREAD

**Discussion by square-foot** [pocketbase/pocketbase#3834] [Q&A] [answered]
Hi
My application creates a file on the server, which the user then has to download. Ideally I wanted the download to start automatically 

I tried using this:   

```
c.response().header().set("Content-Disposition: attachment; filename", "test.bin")
return c.string(200,"Contents of test.bin") 

```
But that did not work 

Finally I got a workaround to create the file in a temporary folder on the server machine using $os.writeFile(...) in a temporary folder and then I constructed an HTML5  which uses the special HTML5 attribute called "download"  for a GET request setup  which downloads the aforesaid file 

The GET route for the file was written like this
(as described here https://pocketbase.io/docs/js-routing#serving-static-files) 

`routerAdd("GET", "/dwnld/*", $apis.staticDirectoryHandler("/home/tmp", false))`

Then the route that is used to download the file was written to send back an HTML where I used the HTML5 anchor element written like this. That worked (except that the user has to consciously click on that link):
```
...
  <a href="/dwnld/test.bin" download="test.bin">Download</a>
...


```

I am wondering why the content-disposition header did not work. Am I doing something incorrectly?

**Comment by ganigeorgiev** [comment, marked as answer]
> My application creates a file on the server

How is this file created? Is it uploaded/saved via the filesystem abstraction, eg. as a record value or it is something totally custom?

Headers should be added using name-value pair notation like `header().set(name, value)`, aka. try with:

```js
c.response().header().set("Content-Disposition", 'attachment; filename="test.bin"')
```

If that doesn't help, please elaborate and/or provide a more detailed code sample.

**Reply by square-foot** [reply]
Ah... that was the problem. I did not write the header correctly. Thank you very much.

The file is ideally to be created dynamically in the route itself. But since my attempt was not working, I tried the other method of temporarily writing the contents of the file to temporary folder and do the above. 

Thank you very much. Much appreciated!
